### PR TITLE
feat(changeset, entity): improved quries performance

### PR DESCRIPTION
**Pull request by melancholiai**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: src/changeset/DAL/typeorm/changesetRepository.ts [hunk 1]
```diff
@@ -36,12 +40,18 @@ export class ChangesetRepository extends Repository<ChangesetDb> implements ICha
...
+  public async tryClosingChangesets(changesetIds: string[], schema: string): Promise<string[]> {
     try {
-      await this.manager.connection.transaction(this.transationIsolationLevel, async (transactionalEntityManager) => {
-        await this.updateFileAsCompleted(changesetIds, schema, transactionalEntityManager);
-
-        await this.updateSyncAsCompleted(changesetIds, schema, transactionalEntityManager);
+      return await this.manager.connection.transaction(this.transationIsolationLevel, async (transactionalEntityManager) => {
+        let completedSyncIds: string[] = [];
+        const completedFilesResult = await this.updateFileAsCompleted(changesetIds, schema, transactionalEntityManager);
+        // check if there are affected rows from the update
```
Comment (syncush) [inline src/changeset/DAL/typeorm/changesetRepository.ts:48]: are there ?

File: src/changeset/models/changesetManager.ts [hunk 2]
```diff
@@ -46,13 +46,13 @@ export class ChangesetManager {
...
 
-  public async closeChangesets(changesetIds: string[]): Promise<void> {
+  public async closeChangesets(changesetIds: string[]): Promise<string[]> {
     if (!this.transactionRetryPolicy.enabled) {
       return this.changesetRepository.tryClosingChangesets(changesetIds, this.dbSchema);
     }
     const retryOptions = { retryErrorType: TransactionFailureError, numberOfRetries: this.transactionRetryPolicy.numRetries as number };
     const functionRef = this.changesetRepository.tryClosingChangesets.bind(this.changesetRepository);
-    await retryFunctionWrapper(retryOptions, functionRef, changesetIds, this.dbSchema);
+    return retryFunctionWrapper(retryOptions, functionRef, changesetIds, this.dbSchema);
```
Comment (syncush) [inline src/changeset/models/changesetManager.ts:55]: I think it should `return await` according to eslint rules, @CptSchnitz  do you recall?

**Review by syncush — APPROVED** [review]

File: src/changeset/DAL/typeorm/changesetRepository.ts (hunk 1 above)
Comment (melancholiai) [reply src/changeset/DAL/typeorm/changesetRepository.ts]: done.

File: src/changeset/models/changesetManager.ts (hunk 2 above)
Comment (melancholiai) [reply src/changeset/models/changesetManager.ts:55]: according to eslint its preferable this way

**Pull request merged**